server/router.js

```diff
@@ -2,22 +2,44 @@ var express = require('express'),
     router = express.Router(),
 
     render = require('./render').render,
-    controllers = require('./controllers');
+    controllers = require('./controllers'),
+    passportGitHub = require('./auth');
+
+// Login routes
+router
+    .get('/auth/github', passportGitHub.authenticate('github', { scope: ['user:email'] }))
+
+    .get('/auth/github/callback', passportGitHub.authenticate('github', { failureRedirect: '/error' }), (req, res) => {
+        res.redirect(req.session.path || '/')
```

Comment:
let's you url query instead of session for return path? e.g. set with `/auth/github/callback/?retpath=req.path` and use as `req.query.retpath || '/'`.